clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -256,11 +256,13 @@ public void poll(long timeout, long now, PollCondition pollCondition) {
      * nor will it execute any delayed tasks.
      */
     public void pollNoWakeup() {
-        disableWakeups();
-        try {
-            poll(0, time.milliseconds(), null);
-        } finally {
-            enableWakeups();
+        synchronized (this) {
```

Comment:
@hachikuji to me passing a parameter to poll would make a lot of sense. Then we could completely drop all the disabling and subsequent enabling of the wake-up and simply wrap the `maybeTriggerWakeup` in an `if`? (I can't see anything wrong with this, but I'm wondering if there wasn't maybe a reason to not do this before? It seems a lot easier than the current solution?)